simplicity_sdk: Import platform/security from Simplicity SDK

asmellby opened the pull request:
Origin: Simplicity SDK
License: Zlib
URL: https://github.com/SiliconLabs/simplicity_sdk
Commit: 8627f8482564dc6d94b56512740a39d6f409a0eb
Version: 2024.12.0
Purpose: Crypto HAL for use by PSA Crypto

jerome-pouiller commented on simplicity_sdk/platform/security/sl_component/sli_psec_osal/inc/sli_psec_osal.h:
```diff
@@ -42,7 +42,10 @@
+#if defined(__ZEPHYR__)
+  #include "sli_psec_osal_zephyr.h"
+  #define SLI_PSEC_THREADING
+#elif defined(SL_CATALOG_MICRIUMOS_KERNEL_PRESENT) || defined(SL_CATALOG_FREERTOS_KERNEL_PRESENT)
```
I suggest have a commit with raw file as they appear in the upstream and another commit with the changes we apply to the upstream (I thought we already did that).

(Maybe the scripts could be in a third commit)

asmellby committed:
scripts: Import platform/security from Simplicity SDK

Update import_simplicity_sdk.py to add platform/security
content.

Signed-off-by: Aksel Skauge Mellbye <aksel.mellbye@silabs.com>

asmellby committed:
simplicity_sdk: Import platform/security from Simplicity SDK

Origin: Simplicity SDK
License: Zlib
URL: https://github.com/SiliconLabs/simplicity_sdk
Commit: 8627f8482564dc6d94b56512740a39d6f409a0eb
Version: 2024.12.0
Purpose: Crypto HAL for use by PSA Crypto

Signed-off-by: Aksel Skauge Mellbye <aksel.mellbye@silabs.com>

asmellby committed:
simplicity_sdk: Patch platform/security content

The OS abstraction layer for the security subsystem must be made
aware of Zephyr.

Signed-off-by: Aksel Skauge Mellbye <aksel.mellbye@silabs.com>

Martinhoff-maker reviewed (approved):
Hard for me to have exhaustive review on this.
